content/struct-classes.md

```diff
@@ -0,0 +1,502 @@
+  override def toString() = s"User($name, $age)"
+  override def hashCode() =
+    MurmurHash3.finalizeHash(
+      MurmurHash3.mix(MurmurHash3.mix(MurmurHash3.productSeed, name), age),
+      2
+    )
+  override def equals(that: Any): Boolean =
+    that match
+      case user: User => user.name == name && user.age == age
+      case _ => false
+
+  def withName(newName: String): User = copy(name = newName)
+  def withAge(newAge: Int): User = copy(age = newAge)
+end User
+~~~
+
+### Specification
+
+A struct class is a `class` definition prefixed with `struct`:
+
+~~~ text
+TmplDef ::= 'struct' 'class' ClassDef
+~~~
+
+A struct class is required to have a parameter list that is not contextual. The
+formal parameters in the first parameter list are called _elements_ and are treated
+specially. A `val` prefix is implicitly added to such a parameter, unless the parameter
+already carries a `val` or `var` modifier. Hence, an accessor definition for the
+parameter is generated.
+
+A struct class definition `c[tps](ps_1)...(ps_n)` with type parameters `tps` and
+value parameters `ps` is handled as follows. A `private` method named `copy` is
+implicitly added to the class definition unless the class already has a member
+(directly defined or inherited) with that name, or the class has a repeated parameter.
+The method is defined as follows:
+
+~~~ scala
+private def copy[Ts](ps_1')...(ps_n'): c[Ts] = c[Ts](xs_1)...(xs_n)
+~~~
+
+Where `Ts` is the vector of types defined in the type parameter section of the class
+definition. Each `xs_i` denotes the parameter names of the parameter section `ps_i`.
+The value parameters `ps_1j'` of the first parameter list have the form
+`x_1j: T_1j = this.x_1j`. The other parameters `ps_ij'` of the `copy` method are
+defined as `x_ij: T_ij`. In all cases, `x_ij` and `T_ij` refer to the name and type
+of the corresponding class parameter `ps_ij`.
+
+Every struct class implicitly overrides some method definitions of class `scala.AnyRef`
+unless a definition of the same method is already given in the struct class itself or
+a concrete definition of the same method is given in some base class of the struct
+class different from `AnyRef`. In particular:
+
+
+- Method `def equals(that: Any): Boolean` is structural equality, where two instances
+  are equal if they both belong to the struct class in question and they have equal
+  (with respect to `equals`) constructor arguments (restricted to the class’s elements,
+  i.e., the first parameter list)
+- Method `def hashCode(): Int` computes a hash-code. If the `hashCode` method of the
+  data structure members map equal (with respect to `equals`) values to equal hash-codes,
+  then the struct class `hashCode` method does too
+- Method `def toString(): String` returns a string representation which contains the
+  name of the class and its elements.
+
+Finally, mirrors (instances of `scala.deriving.Mirror`) are not synthesized by the
```

Comment:
Could you elaborate on why `struct` classes won't get synth Mirror instances? 
If there are bin compact reasons or another rationale why `struct` classes don't get these may be good to have them explained here.

Currently, Scala 3 docs at https://docs.scala-lang.org/scala3/reference/contextual/derivation.html#mirror say

> Instances of the Mirror type class are generated automatically by the compiler unconditionally for:
> enums and enum cases,
> case objects.

> Instances for Mirror are also generated conditionally for:
> case classes where the constructor is visible at the callsite (always true if the companion is not a case object)
> sealed classes and sealed traits where:
> there exists at least one child case,
> each child case is reachable from the parent's definition,
> if the sealed trait/class has no companion, then each child case is reachable from the callsite through the prefix of the type being mirrored,
> and where the compiler can generate a Mirror type class instance for each child case.

Reply:
If we synthesize a `Mirror`, we get static access to its `MirroredElemTypes` type member. When you then add a field, the type behind that type alias changes in a binary-incompatible way.

Reply:
I have clarified the motivation for not having mirrors in 6a9fdee